src/python/pants/engine/internals/graph.py

```diff
@@ -386,6 +387,18 @@ async def find_owners(owners_request: OwnersRequest) -> Owners:
             matching_files = set(
                 matches_filespec(candidate_tgt.get(Sources).filespec, paths=sources_set)
             )
+            # Also consider secondary ownership, meaning it's not a `Sources` field with primary
+            # ownership, but the target still should match the file. We can't use `tgt.get()`
+            # because this is a mixin, and there technically may be >1 field.
+            secondary_owner_fields = tuple(
+                field  # type: ignore[misc]
```

Comment:
Weird MyPy error - I didn't spend time trying to figure it out:

```
12:34:58.20 [WARN] Completed: typecheck - MyPy failed (exit code 1).
src/python/pants/engine/internals/graph.py: note: In function "find_owners":
src/python/pants/engine/internals/graph.py:394:17: error: Generator has
incompatible item type
"pants.engine.internals.graph.<subclass of "Field" and "SecondaryOwnerMixin">1";
expected
"pants.engine.internals.graph.<subclass of "Field" and "SecondaryOwnerMixin">"
[misc]
                    field
                    ^
Found 1 error in 1 file (checked 421 source files)
```

Reply:
This is where I wish there was a sad trombone emoji.